### PR TITLE
[crypto] Fix serialization/validation TODOs in bls12381

### PR DESCRIPTION
**Pull request by kchalkias**
## Motivation

Addressing various TODOs in BLS implementation related to serialization and key validation.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

YES

## Test Plan

`bls12381_test::test_keys_custom_serialisation()`

### CODE REVIEW

File: crypto/nextgen_crypto/src/bls12381.rs [hunk 1]
```diff
@@ -43,56 +43,44 @@ use pairing::{
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 use std::ops::Deref;
-use threshold_crypto;
 
-// type alias for this unwieldy type
+/// The length of the Ed25519PrivateKey.
+pub const BLS12381_PRIVATE_KEY_LENGTH: usize = 32;
```
Comment (huitseeker) [inline crypto/nextgen_crypto/src/bls12381.rs]: Isn't it this: https://github.com/poanetwork/threshold_crypto/blob/485333db6e2611d26a2a921a6ee1e1dcc3ce4624/src/secret.rs#L11
given :
https://github.com/poanetwork/threshold_crypto/blob/master/src/lib.rs#L336
?

File: crypto/nextgen_crypto/src/bls12381.rs [hunk 2]
```diff
@@ -149,16 +137,16 @@ impl TryFrom<&[u8]> for BLS12381PrivateKey {
...
+        // This is expected to succeed as we just return the bytes of an existing key.
+        bincode::serialize(&self.0).unwrap()
     }
 }
 
 impl Genesis for BLS12381PrivateKey {
     fn genesis() -> Self {
-        let mut buf = [0u8; threshold_crypto::PK_SIZE];
-        buf[threshold_crypto::PK_SIZE - 1] = 1;
+        let mut buf = [0u8; BLS12381_PUBLIC_KEY_LENGTH];
```
Comment (huitseeker) [inline crypto/nextgen_crypto/src/bls12381.rs]: Genesis returns the private key, not the public key right?

Comment (kchalkias) [reply crypto/nextgen_crypto/src/bls12381.rs]: Nice catch, oops it originally mentioned `PK_SIZE`, hm,  this is a fix then. Thanks!

File: crypto/nextgen_crypto/src/bls12381.rs (hunk 1 above)
Comment (kchalkias) [reply crypto/nextgen_crypto/src/bls12381.rs]: True, but I as I don't have access on it `pub(crate)`, I fix it to 32, as in BLS12_381 `r` is a 255 bits, see [BLS12-381 specs](https://electriccoin.co/blog/new-snark-curve/).

**Pull request merged**